src/platforms/web/compiler/directives/model.js

```diff
@@ -4,6 +4,7 @@ import config from 'core/config'
 import { addHandler, addProp, getBindingAttr } from 'compiler/helpers'
 import { genComponentModel, genAssignmentCode } from 'compiler/directives/model'
 
+const ternaryTextInputRE = /^[^?]{1,}\?\s*('|"|`)(?:text|number|password|search|email|tel|url)\1\s*:\s*('|"|`)(?:text|number|password|search|email|tel|url)\2\s*$/
```

Comment:
Shouldn't use [makeMap](https://github.com/vuejs/vue/blob/dev/src/shared/util.js#L87) be enough? Don't we want to check if dynamic type is one of those?